src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[10]; // Ініціалізація масиву з початковим розміром
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            resize(); // Якщо масив заповнений, збільшуємо розмір
+        }
+        elements[size++] = value; // Додаємо елемент і збільшуємо розмір
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
```

Comment:
The index validation logic is repeated in multiple methods. Consider refactoring this logic into a separate private method to adhere to the DRY principle as suggested in the checklist.